sa/sa_test.go

```diff
@@ -1419,21 +1006,36 @@ func TestSetOrderProcessing(t *testing.T) {
 		RegistrationID: reg.ID,
 		Status:         core.StatusPending,
 		Expires:        &authzExpires,
+		Challenges: []core.Challenge{
+			{
+				Token:  "YXNk",
+				Type:   core.ChallengeTypeDNS01,
+				Status: core.StatusPending,
+			},
+		},
 	}
-	authz, err := sa.NewPendingAuthorization(ctx, newAuthz)
-	test.AssertNotError(t, err, "Couldn't create new pending authorization")
+	authzPB, err := bgrpc.AuthzToPB(newAuthz)
+	test.AssertNotError(t, err, "bgrpc.AuthzToPB failed")
+	ids, err := sa.NewAuthorizations2(ctx, &sapb.AddPendingAuthorizationsRequest{Authz: []*corepb.Authorization{authzPB}})
```

Comment:
Similar to the RA I think the process of creating a new authz2 and then finalizing it at the SA level is duplicated a fair amount in the tests. Can a helper function simplify some of the duplication?